tools/msm_execute_helper.c

```diff
@@ -8,10 +8,12 @@ int main(int argc, char* argv[])
     gchar **argv_modified = g_new0 (gchar *, argc + 1);
```

Comment:
`gchar** argv_modified` is a pointer to the allocated memory by `g_new0 (gchar *, argc + 1);`, which stores argc + 1 `gchar*` pointers.

https://github.com/mate-desktop/mate-system-monitor/blob/65cff87dca39dd84359c47619234a44f2e0467f8/src/procman_pkexec.cpp#L12-L13
For instance:
- PKGLIBEXECDIR = "/usr/libexec/mate-system-monitor"
- command = "kill -s 9 26304"
```
"pkexec --disable-internal-agent /usr/libexec/mate-system-monitor/msm-kill -s 9 26304"
```

https://github.com/mate-desktop/mate-system-monitor/blob/65cff87dca39dd84359c47619234a44f2e0467f8/tools/msm_execute_helper.c#L6
argc:=4, argv:
- argv[0] := "/usr/libexec/mate-system-monitor/msm-kill"
- argv[1] := "-s"
- argv[2] := "9"
- argv[3] := "26304"

https://github.com/mate-desktop/mate-system-monitor/blob/65cff87dca39dd84359c47619234a44f2e0467f8/tools/msm_execute_helper.c#L9-L10
argv_modified:
- argv_modified[0] := COMMAND,  COMMAND := "kill"
- argv[1] := "-s"
- argv[2] := "9"
- argv[3] := "26304"
- argv[4] := 0

Reply:
OK, I  can see where  
` g_free (argv_modified);`
is needed, and that that would fix the memory leak
What I don't understand is where `errno` is set in the first place, but it's presence predates this PR and thus should not hold up the merge.. Beyond my not understanding that, the memory leak fix looks good and I've already confirmed the change creates no problems when tested on my system